SDK 50 and LinearGradientView does not work with start and end coordinates

### Minimal reproducible example

https://snack.expo.dev/@hnzufx/simple-linear-gradient

### Summary

In Expo SDK 50 you cannot set start and end coordinates. You will get the error:

 ERROR  Cannot set prop 'endPoint' on view 'class expo.modules.lineargradient.LinearGradientView'
→ Caused by: java.lang.IllegalArgumentException: Class declares 0 type parameters, but 2 were provided.
 ERROR  Cannot set prop 'startPoint' on view 'class expo.modules.lineargradient.LinearGradientView'
→ Caused by: java.lang.IllegalArgumentException: Class declares 0 type parameters, but 2 were provided.

You can reproduce the bug by creating a new project:

npx expo install expo-linear-gradient
cd expo-linear-gradient

and the linked snack as App.js. The snack works because it's SDK 49. In SDK 50 it doesn't work.


### Environment

  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 13.5.2
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.11.0 - /usr/local/opt/node@20/bin/node
      Yarn: 1.22.21 - /usr/local/bin/yarn
      npm: 10.2.4 - /usr/local/opt/node@20/bin/npm
      Watchman: 2023.12.04.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.14.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, visionOS 1.0, watchOS 10.2
    IDEs:
      Android Studio: 2021.1 AI-211.7628.21.2111.8193401
      Xcode: 15.2/15C500b - /usr/bin/xcodebuild
    npmPackages:
      expo: ~50.0.2 => 50.0.2 
      react: 18.2.0 => 18.2.0 
      react-native: 0.73.2 => 0.73.2 
    npmGlobalPackages:
      eas-cli: 6.0.0
      expo-cli: 6.3.10
    Expo Workflow: managed

--- a/packages/expo-linear-gradient/CHANGELOG.md
+++ b/packages/expo-linear-gradient/CHANGELOG.md
@@ -10,6 +10,8 @@
 
 ### 💡 Others
 
+- On `Android`, remove type annotation on `View`.
+
 ## 12.7.0 — 2023-11-14
 
 ### 🛠 Breaking changes

--- a/packages/expo-linear-gradient/CHANGELOG.md
+++ b/packages/expo-linear-gradient/CHANGELOG.md
@@ -10,7 +10,7 @@
 
 ### 💡 Others
 
-- On `Android`, remove type annotation on `View`.
+- On `Android`, remove type annotation on `View`. ([#26545](https://github.com/expo/expo/pull/26545) by [@alanjhughes](https://github.com/alanjhughes))
 
 ## 12.7.0 — 2023-11-14
 

--- a/packages/expo-linear-gradient/android/src/main/java/expo/modules/lineargradient/LinearGradientModule.kt
+++ b/packages/expo-linear-gradient/android/src/main/java/expo/modules/lineargradient/LinearGradientModule.kt
@@ -3,32 +3,31 @@ package expo.modules.lineargradient
 import expo.modules.kotlin.modules.Module
 import expo.modules.kotlin.modules.ModuleDefinition
 
-typealias ViewType = LinearGradientView
 
 class LinearGradientModule : Module() {
   override fun definition() = ModuleDefinition {
     Name("ExpoLinearGradient")
 
     View(LinearGradientView::class) {
-      Prop("colors") { view: ViewType, colors: IntArray ->
+      Prop("colors") { view, colors: IntArray ->
         view.setColors(colors)
       }
 
-      Prop("locations") { view: ViewType, locations: FloatArray? ->
+      Prop("locations") { view, locations: FloatArray? ->
         locations?.let {
           view.setLocations(it)
         }
       }
 
-      Prop("startPoint") { view: ViewType, startPoint: Pair<Float, Float>? ->
+      Prop("startPoint") { view, startPoint: Pair<Float, Float>? ->
         view.setStartPosition(startPoint?.first ?: 0.5f, startPoint?.second ?: 0f)
       }
 
-      Prop("endPoint") { view: ViewType, endPoint: Pair<Float, Float>? ->
+      Prop("endPoint") { view, endPoint: Pair<Float, Float>? ->
         view.setEndPosition(endPoint?.first ?: 0.5f, endPoint?.second ?: 1f)
       }
 
-      Prop("borderRadii") { view: ViewType, borderRadii: FloatArray? ->
+      Prop("borderRadii") { view, borderRadii: FloatArray? ->
         view.setBorderRadii(borderRadii ?: FloatArray(8) { 0f })
       }
     }

--- a/packages/expo-linear-gradient/android/src/main/java/expo/modules/lineargradient/LinearGradientModule.kt
+++ b/packages/expo-linear-gradient/android/src/main/java/expo/modules/lineargradient/LinearGradientModule.kt
@@ -3,7 +3,6 @@ package expo.modules.lineargradient
 import expo.modules.kotlin.modules.Module
 import expo.modules.kotlin.modules.ModuleDefinition
 
-
 class LinearGradientModule : Module() {
   override fun definition() = ModuleDefinition {
     Name("ExpoLinearGradient")

--- a/packages/expo-modules-core/CHANGELOG.md
+++ b/packages/expo-modules-core/CHANGELOG.md
@@ -13,6 +13,7 @@
 ### 🐛 Bug fixes
 
 - Fixed breaking changes from React-Native 0.74. ([#26357](https://github.com/expo/expo/pull/26357) by [@kudo](https://github.com/kudo))
+- Fix proguard rules so `Serializable` types are not obfuscated.
 
 ### 💡 Others
 

--- a/packages/expo-modules-core/CHANGELOG.md
+++ b/packages/expo-modules-core/CHANGELOG.md
@@ -13,7 +13,7 @@
 ### 🐛 Bug fixes
 
 - Fixed breaking changes from React-Native 0.74. ([#26357](https://github.com/expo/expo/pull/26357) by [@kudo](https://github.com/kudo))
-- Fix proguard rules so `Serializable` types are not obfuscated.
+- Fix proguard rules so `Serializable` types are not obfuscated. ([#26545](https://github.com/expo/expo/pull/26545) by [@alanjhughes](https://github.com/alanjhughes))
 
 ### 💡 Others
 

--- a/packages/expo-modules-core/android/proguard-rules.pro
+++ b/packages/expo-modules-core/android/proguard-rules.pro
@@ -9,6 +9,7 @@
 -keep enum * implements expo.modules.kotlin.types.Enumerable {
   *;
 }
+-keepnames class * implements java.io.Serializable
 
 -keep,allowoptimization,allowobfuscation class * extends expo.modules.kotlin.modules.Module {
   public <init>();

--- a/packages/expo-modules-core/android/proguard-rules.pro
+++ b/packages/expo-modules-core/android/proguard-rules.pro
@@ -9,7 +9,7 @@
 -keep enum * implements expo.modules.kotlin.types.Enumerable {
   *;
 }
--keepnames class * implements java.io.Serializable
+-keepnames class kotlin.Pair
 
 -keep,allowoptimization,allowobfuscation class * extends expo.modules.kotlin.modules.Module {
   public <init>();

